token-lending/program/src/processor.rs

```diff
@@ -1874,6 +1943,57 @@ fn process_flash_loan(
     Ok(())
 }
 
+#[inline(never)] // avoid stack frame limit
+fn process_withdraw_obligation_collateral_and_redeem_reserve_liquidity(
+    program_id: &Pubkey,
+    collateral_amount: u64,
+    accounts: &[AccountInfo],
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let reserve_collateral_info = next_account_info(account_info_iter)?;
+    let user_collateral_info = next_account_info(account_info_iter)?;
+    let reserve_info = next_account_info(account_info_iter)?;
+    let obligation_info = next_account_info(account_info_iter)?;
+    let lending_market_info = next_account_info(account_info_iter)?;
+    let lending_market_authority_info = next_account_info(account_info_iter)?;
+    let user_liquidity_info = next_account_info(account_info_iter)?;
```

Comment:
Should call it `destination_liquidity_info` to be consistent with how `_redeem_reserve_collateral` refers to it.

Also for whatever reason, a user may not want to withdraw to themselves. We see this with proxy wallets and gassless txs in Ethereum. Basically the naive way to implement in Solidity is to withdraw to the caller (`msg.sender`), but for gasless txs the user signs a message that the smart contract verifies, and a _relayer_ submits the tx (so they're the `msg.sender`). The result is an improved UX where the user signs something but doesn't have to spend the gas, and maybe doesn't even need a web3 wallet like Metamask at all.

Anyway just some Ethereum history.